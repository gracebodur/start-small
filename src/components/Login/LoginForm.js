import React from 'react'
import { Link } from 'react-router-dom'
import './LoginForm.css'

function LoginForm({ onRouteChange}) {
    return (
        <main className='Login-main center'>
            <form className= 'Login-form center'>
                <fieldset className='Login-fieldset'>
                    <legend 
                        className='Login-legend center'>
                        User Log In
                    </legend>
                    <div className="Login-user_name">
                    <label 
                        className="Username-label" 
                        htmlFor="login-username">
                        Username
                    </label>
                    <input
                        id="login-username" 
                        className="Username-input" 
                        type="text" 
                        name="user_name"
                        placeholder="Enter Username" />
                    </div>
                    <div className="Login-password">
                    <label 
                        className='Password-label' 
                        htmlFor="login-password">
                        Password
                    </label>
                    <input 
                        className='Password-input' 
                        type="password" 
                        name="password" 
                        id="login-password" 
                        placeholder="Enter Password" />
                    </div>
                    <div className="Login-button center">
                   <Link to='/users-page'><input
                        className="Login-button-input" 
                        type="submit" 
                        value="Log in"/></Link>
                    </div>
                    <div className="Cancel-button center">
                    <Link to='/search-page'><input 
                        className="Login-cancel-button-input" 
                        type="submit" 
                        value="Cancel"/></Link>
                    </div>
                    <div className="Register-link center">
                    <Link to='/users-page'
                        className="Login-register-link" >
                        Register</Link>
                    </div>
              </fieldset>
            </form>
        </main>
    )
}

export default LoginForm