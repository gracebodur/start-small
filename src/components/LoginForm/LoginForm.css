.Login-main {
    border-radius: .25rem;
    border-style: solid;
    border-width: 1px;
    color: var( --clr-purple);
    border-color: rgba(0, 0, 0, .1);
    margin-bottom: 3rem;
    max-width: 16rem;
    padding: 2rem;
    border-radius: 10px;
    letter-spacing: 2px;
    -webkit-box-shadow: -11px 10px 6px -1px rgba(54,34,69,0.72);
    -moz-box-shadow: -11px 10px 6px -1px rgba(54,34,69,0.72);
    box-shadow: -11px 10px 6px -1px rgba(54,34,69,0.72);
    background-color: #12CAD6;
}

.Login-form {
    max-width: 30em;
}

.Login-fieldset {
    text-transform: uppercase;
    border-style: solid;
    border-width: 1px;
    border-color: transparent;
    padding-left: 0;
    padding-right: 0;
    margin-left: 0;
    margin-right: 0;
}

.Login-legend {
    font-size: 1.25rem;
    font-weight: 600;
    padding-left: 0;
    padding-right: 0;
    margin-left: 0;
    margin-right: 0;
}

.Login-user_name {
    margin-top: 1rem;
}

.Username-label,
.Password-label {
    display: block;
    font-weight: 600;
    line-height: 1.5;
    font-size: .875rem;
}

.Username-input,
.Password-input {
    padding: .5rem;
    -webkit-appearance: none;
    -moz-appearance: none;
    border-style: solid;
    border-width: 1px;
    background-color: transparent;
    width: 100%;
    border-radius: 10px;
}

.Username-input:hover {
    background-color: #0FABBC;
}


.Password-input:hover {
    background-color: #0FABBC;
}


.Login-button,
.Cancel-button {
    line-height: 1.5;
    margin-top: 1rem;
}

.Login-button-input,
.Login-cancel-button-input {
    display: inline-block;
    font-weight: bold;
    font-size: .900rem;
    padding-left: 1rem;
    padding-right: 1rem;
    padding-top: .5rem;
    padding-bottom: .5rem;
    -webkit-appearance: none;
    -moz-appearance: none;
    border-style: solid;
    border-width: 1px;
    border-color:#222831;
    border-radius: 10px;
    text-transform: uppercase;
    background-color: transparent;
     color: var( --clr-purple);
    -moz-osx-font-smoothing: grayscale;
    backface-visibility: hidden;
    transform: translateZ(0);
    transition: transform .25s ease-out;
}


/* iphone6/7/8*/
@media only screen 
  and (min-device-width: 375px) 
  and (max-device-width: 667px) 
  and (-webkit-min-device-pixel-ratio: 2) { 
    .Login-main {
        margin-top: 30%;
    }
}


/* iphoneX */
@media only screen 
  and (min-device-width: 375px) 
  and (max-device-width: 812px) 
  and (-webkit-min-device-pixel-ratio: 3) { 
    .Login-main {
        margin-top: 30%;
    }
}

/* tablet */
@media only screen 
  and (min-device-width: 400px) 
  and (max-device-width: 568px) 
  and (-webkit-min-device-pixel-ratio: 2) {
    .Login-main {
        margin-top: 30%;
        max-width: 15rem;
    }
    .Login-legend {
        font-size: 1rem;
    }
    .Username-label {
        font-size: 1rem;
    }
    .Password-label {
        font-size: 1rem;
    }
    .Username-input,
    .Password-input {
        font-size: 1rem;
    }
    .Login-button {
        max-width: 20rem;
    }
    .Cancel-button {
        max-width: 20rem;
    }
}
    

/* ipad */
@media only screen 
  and (min-device-width: 768px) 
  and (max-device-width: 1024px) 
  and (-webkit-min-device-pixel-ratio: 2) {
    .Login-main {
        margin-top: 20%;
        max-width: 20rem;
    }
    .Login-legend {
        font-size: 2.0rem;
        font-weight: 600;
    }
    .Username-label {
        font-size: 1.25rem;
    }
    .Password-label {
        font-size: 1.25rem;
    }
    .Username-input,
    .Password-input {
        font-size: 1rem;
    }
    .Login-button {
        max-width: 30rem;
    }
    .Cancel-button {
        max-width: 30rem;
    }
}

/* ipadpro */
@media only screen 
  and (min-device-width: 834px) 
  and (max-device-width: 1112px)
  and (-webkit-min-device-pixel-ratio: 2) {
    .Login-button-input {
        font-size: 1.5rem;
    }
    .Login-cancel-button-input {
        font-size: 1.5rem;
    }
}
