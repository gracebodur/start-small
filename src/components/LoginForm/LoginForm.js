import React, { Component } from 'react'
import TokenService from '../../services/token-service'
import AuthApiService from '../../services/auth-api-service'
import { Section } from '../../components/Utils/Utils'
import './LoginForm.css'

class LoginForm extends Component {
    static defaultProps = {
        onLoginSuccess: () => { }
    }

    constructor ( props ) {
        super( props )
        this.state = {
            user_name: '',
            password: '',
            error: null
        }
    }

    handleSubmitBasicAuth = ev => {
        ev.preventDefault()
        const { user_name, password } = ev.target

        TokenService.saveAuthToken(
            TokenService.makeBasicAuthToken( user_name.value, password.value )
        )
        user_name.value = ''
        password.value = ''
        this.props.onLoginSuccess()
    }

    handleSubmitJwtAuth = ev => {
        ev.preventDefault()
        this.setState( { error: null } )
        const { user_name, password } = ev.target

        AuthApiService.postLogin( {
            user_name: user_name.value,
            password: password.value,
        } )
            .then( res => {
                user_name.value = ''
                password.value = ''
                TokenService.saveAuthToken( res.authToken )
                this.props.onLoginSuccess()
            } )
            .catch( res => {
                this.setState( { error: res.error } )
            } )
    }

    render () {
        const { error } = this.state
        return (
            <Section className='container'>
                <div className="login-form center_div row">
                    <form onSubmit={this.handleSubmitJwtAuth} className='col-md-12 form-wrapper'>
                        <h2 className='login-header h2'>Login</h2>
                        <div role='alert'>
                            {error && <p className='red'>{error}</p>}
                        </div>
                        <div className="form-group">
                            <label className='h6'>Username</label>
                            <input
                                type="text"
                                className="form-control form-input"
                                id="form-username"
                                name="user_name"
                                placeholder="Type your username"
                                required />
                        </div>
                        <div className="form-group">
                            <label className='h6'>Password</label>
                            <input
                                type="password"
                                className="form-control form-input"
                                id="form-password"
                                name="password"
                                placeholder="&#9679;&#9679;&#9679;&#9679;&#9679;"
                                required />
                        </div>
                        <div className="form-group clearfix">
                            <button type="submit" className="btn float-left">Sign in</button>
                        </div>
                        
                    </form>
                </div>
            </Section>
        )
    }

}

export default LoginForm


