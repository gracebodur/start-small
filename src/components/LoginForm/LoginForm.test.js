import React from 'react';
import ReactDOM from 'react-dom';
import LoginForm from './LoginForm';
import { BrowserRouter as Router } from 'react-router-dom';

describe('Log in Form Component', () => {
    it('renders without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(<Router> 
                        <LoginForm /> 
                    </Router>, div);
      ReactDOM.unmountComponentAtNode(div);
    });
});