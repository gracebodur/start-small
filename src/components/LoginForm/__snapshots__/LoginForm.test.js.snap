// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm component renders a LoginForm by default 1`] = `undefined`;

exports[`LoginForm component renders the LoginForm given props 1`] = `undefined`;
